decidim-regulations/app/services/decidim/regulations/regulation_search.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Regulations
+    # Service that encapsulates all logic related to filtering participatory processes.
+    class RegulationSearch < ParticipatorySpaceSearch
+      def initialize(options = {})
+        super(ParticipatoryProcess.all, options).where(decidim_participatory_process_group_id: Rails.application.config.regulation)
```

Comment:
I don't really understand this initializer. It may work, but just by accident?
Is this `where` really assigned to something? or the object looses it after initialization?

Shouldn't this `where` be setted in the `base_query`?

Reply:
This is not an initializer, this is a service for filtering participatory_processes.

What do you mean with "It may work, but just by accident?"

`RegulationSearch` is used in `RegulationFiltersCell` and `RegulationController` and is used for filtering processes with regulations process_group..

Are you suggesting to do this?
```suggestion
def initialize(options = {})
   super(ParticipatoryProcess.all, options)
end

def base_query
   super.where(decidim_participatory_process_group_id: Rails.application.config.regulation)
end
```

Reply:
You are right @isaacmg410 I meant constructor instead of initializer :D and the fact that you don't really know what a super-constructor may return now or in future changes. Also, you are adding a `where` clause but it is not being assigned anywhere, what happens with this `where` is just casual.

Yes, what you suggest is what I meant

Reply:
Requested changes applied! :)